ipn/ipnlocal/peerapi.go

```diff
@@ -51,9 +50,14 @@ var initListenConfig func(*net.ListenConfig, netip.Addr, *interfaces.State, stri
 // ("cleartext" HTTP/2) support to the peerAPI.
 var addH2C func(*http.Server)
 
+// peerDNSQueryer is implemented by tsdns.Resolver.
+type peerDNSQueryer interface {
```

Comment:
nit: naming. `peerDNSQueryer` vs `peerAPIServer` is giving fuzzy/confusing star signs. Maybe something with `backend` in it?

Reply:
`peerDNSQueryHandler` mimics the underlying method name. The naming convention here is `-er` suffix for single method interfaces. `Backend` would be confusing, as backend exists in the ipn packages, but not in the tsdns packages.